specification/requirements_model/model_rules/columns/effectivecost.json

```
@@ -0,0 +1,412 @@
+    "Function": "Nullability",
+    "Reference": "EffectiveCost",
+    "EntityType": "Column",
+    "Notes": "",
+    "CRVersionIntroduced": "1.2",
+    "Status": "Active",
+    "ApplicabilityCriteria": [],
+    "Type": "Static",
+    "ValidationCriteria": {
+      "MustSatisfy": "EffectiveCost MUST NOT be null.",
+      "Keyword": "MUST NOT",
+      "Requirement": {
+        "CheckFunction": "CheckNotValue",
+        "ColumnName": "EffectiveCost",
+        "Value": null
+      },
+      "Condition": {},
+      "Dependencies": []
+    }
+  },
+  "EffectiveCost-C-004-M": {
+    "Function": "Validation",
+    "Reference": "EffectiveCost",
+    "EntityType": "Column",
+    "Notes": "",
+    "CRVersionIntroduced": "1.2",
+    "Status": "Active",
+    "ApplicabilityCriteria": [],
+    "Type": "Static",
+    "ValidationCriteria": {
+      "MustSatisfy": "EffectiveCost MUST be a valid decimal value.",
+      "Keyword": "MUST",
+      "Requirement": {
+        "CheckFunction": "CheckNotValue",
+        "ColumnName": "ConsumedQuantity",
+        "Value": null
+      },
+      "Condition": {},
+      "Dependencies": []
+    }
+  },
+  "EffectiveCost-C-005-C": {
+    "Function": "Validation",
+    "Reference": "EffectiveCost",
+    "EntityType": "Column",
+    "Notes": "",
+    "CRVersionIntroduced": "1.2",
+    "Status": "Active",
+    "ApplicabilityCriteria": [],
+    "Type": "Static",
+    "ValidationCriteria": {
+      "MustSatisfy": "EffectiveCost MUST be 0 when ChargeCategory is \"Purchase\" and the purchase is intended to cover future eligible charges.",
+      "Keyword": "MUST",
+      "Requirement": {
+        "CheckFunction": "CheckValue",
+        "ColumnName": "EffectiveCost",
+        "Value": 0
+      },
+      "Condition": {
+        "CheckFunction": "CheckValue",
+        "ColumnName": "ChargeCategory",
+        "Value": "Purchase"
+      },
+      "Dependencies": [
+        "ChargeCategory-C-000-M"
+      ]
+    }
+  },
+  "EffectiveCost-C-006-M": {
+    "Function": "Validation",
+    "Reference": "EffectiveCost",
+    "EntityType": "Column",
+    "Notes": "Cross-column reference: BillingCurrency",
+    "CRVersionIntroduced": "1.2",
+    "Status": "Active",
+    "ApplicabilityCriteria": [],
+    "Type": "Dynamic",
+    "ValidationCriteria": {
+      "MustSatisfy": "EffectiveCost MUST be denominated in the BillingCurrency.",
+      "Keyword": "MUST",
+      "Requirement": {},
+      "Condition": {},
+      "Dependencies": [
+        "BillingCurrency-C-000-M"
+      ]
+    }
+  },
+  "EffectiveCost-C-007-O": {
+    "Function": "Validation",
+    "Reference": "EffectiveCost",
+    "EntityType": "Column",
+    "Notes": "",
+    "CRVersionIntroduced": "1.2",
+    "Status": "Active",
+    "ApplicabilityCriteria": [],
+    "Type": "Dynamic",
+    "ValidationCriteria": {
+      "MustSatisfy": "The sum of EffectiveCost in a given billing period MAY not match the sum of the invoices received for the same billing period for a billing account.",
```

Although the normative requirement in section 1.2 currently states:

> * The sum of EffectiveCost in a given billing period **may not** match the sum of the invoices received for the same billing period for a billing account.

I suggest we go beyond simply rephrasing "...may not match..." to "...MAY not match..." since the original phrasing is ambiguous - can be interpreted either as a possibility ("might not") or as a soft prohibition ("is not allowed"), which is unclear in a normative context.

BTW, added [this](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/1380#issuecomment-3383522798) comment in FR #1380 to ensure this is addressed in the column definition as soon as possible.

```suggestion
      "MustSatisfy": "The sum of EffectiveCost in a given billing period MAY differ from the sum of the invoices received for the same billing period for a billing account.",
```

---

I do not think we should differ in language from the 1.2 spec. "Fixing" the language in the CRs isn't resolving the issue of ambiguity.

---

Discussed in maintainers. agreed to keep as is.